site/app/models/DockerUI.php

```diff
@@ -0,0 +1,501 @@
+    private array $json_data;
+    /** List of worker machine names */
+    protected array $worker_machines;
+    /** List of capability tags */
+    protected array $capabilities;
+    /** List of capabilities unassociated with any container */
+    protected array $no_image_capabilities;
+    /** Mapping between capabilities and the color to draw their badge */
+    protected array $capability_to_color_mapping;
+    /** Human readable string of when the docker logs generated by Submitty were last parsed */
+    protected string $last_ran = "Unknown";
+    /** Human readable string of when the sysinfo logs were last parsed */
+    protected string $sysinfo_last_updated = "unknown";
+    /** List of errors from parsing log files  */
+    protected array $error_logs = [];
+    /** location of where the sysinfo log files generated by Submitty are located */
+    private string $sysinfo_filepath;
+    /** location of where logs generated by Submitty related to docker are located */
+    private string $docker_logpath;
+    /** A list of \DockerImages */
+    protected array $docker_images = [];
+    private array $fail_images;
+    /** Mapping between images to which Submitty capabilities they are associated with */
+    private array $image_to_capability_mapping;
+    /** 
+     * Mapping between workers and the docker version they run, this is collected from the 
+     * docker logs but displayed later in the worker machines table
+     */
+    private array $worker_docker_versions = [];
+    /** 
+     * Mapping between workers and the os they have, this is collected from the 
+     * docker logs but displayed later in the worker machines table
+     */
+    private array $worker_os_names = [];
+
+
+    /** Create a new docker UI object from the json data read from the filesystem */
+    public function __construct(Core $core, array $json) {
+        parent::__construct($core);
+
+        /** Sanity checks on given json structure */
+        assert(array_key_exists('autograding_workers', $json) && is_array($json['autograding_workers']));
+        assert(array_key_exists('autograding_containers', $json) && is_array($json['autograding_containers']));
+
+        $this->sysinfo_filepath = FileUtils::joinPaths($this->core->getConfig()->getSubmittyPath(), "logs", "sysinfo");
+        $this->docker_logpath = FileUtils::joinPaths($this->core->getConfig()->getSubmittyPath(), "logs", "docker");
+
+        $this->json_data = $json;
+
+        try {
+            $this->parseWorkersAndCapabilities($this->json_data['autograding_containers'], $this->json_data['autograding_workers']);
+            $this->parseDockerLogs();
+            $this->parseSystemInformationLogs();
+        } catch (Exception $e) {
+            $this->error_logs[] = $e->getMessage();
+            Logger::error($e->getMessage());
+        }
+    }
+
+    /** Return a list of autograding worker names */
+    public function getAutogradingContainers(): array {
```

Comment:
What's the expected array structure here?  You can add a PHPDoc comment to specify the array type for better static analysis results.